azext_iot/iothub/providers/message_route.py

```diff
@@ -0,0 +1,184 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from typing import Optional
+from knack.log import get_logger
+from azure.cli.core.azclierror import ResourceNotFoundError
+from azext_iot.common.utility import process_json_arg
+from azext_iot.iothub.common import RouteSourceType
+from azext_iot.iothub.providers.base import IoTHubResourceProvider
+
+
+logger = get_logger(__name__)
+
+
+class MessageRoute(IoTHubResourceProvider):
+    def __init__(
+        self,
+        cmd,
+        hub_name: str,
+        rg: Optional[str] = None,
+    ):
+        super(MessageRoute, self).__init__(cmd, hub_name, rg)
+
+    def create(
+        self,
+        route_name: str,
+        source_type: str,
+        endpoint_name: str,
+        enabled: bool = True,
+        condition: Optional[str] = None,
```

Comment:
cant we just put str = 'true' here?

Reply:
May need to test this, but if `None` is passed into `create()` - I think you still need to check `if not condition` or `if condition is None` later - because the default parameter only gets used if there is no value passed to the function, `None` is still considered a value iirc

Reply:
yeah this is the check:
`"condition": ('true' if condition is None else condition),`